src/precice/impl/DataContext.hpp

```diff
@@ -71,16 +71,6 @@ class DataContext {
    */
   void mapData();
 
-  /**
-   * @brief Adds a MappingContext and the MeshContext required by the mapping to the corresponding DataContext data structures.
-   *
-   * A mapping maps the given data from or to _providedData (depending on whether it is a read or write mapping).
-   *
-   * @param[in] mappingContext Context of the mapping
-   * @param[in] meshContext Context of mesh this mapping is mapping from or to
-   */
-  virtual void appendMappingConfiguration(MappingContext &mappingContext, const MeshContext &meshContext) = 0;
```

Comment:
removed because Global read/write DataContexts inherit from `DataContext`, but don't implement `appendMappingConfiguration(...)`

Reply:
But then you have to declare this function in `ReadDataContext` and `WriteDataContext`. We could, of course, introduce an intermediate layer `LocalDataContext` just for this purpose, but this looks overengineered. You can also just implement it in `Read/WriteGlobalDataContext` and trigger an assertion, if somebody tries to call the function.

In the end this is a hint for some problem with the class design, but I think we can live with it.